sotodlib/toast/workflows/proc_mapmaker.py

```diff
@@ -143,35 +143,58 @@ def mapmaker(job, otherargs, runargs, data):
         if job_tmpls.baselines.enabled:
             job_tmpls.noise_model = noise_model
 
-        if otherargs.obsmaps:
-            # Map each observation separately
-            timer_obs = toast.timing.Timer()
-            timer_obs.start()
-            group = data.comm.group
-            orig_name = job_ops.mapmaker.name
-            orig_comm = data.comm
-            new_comm = toast.Comm(world=data.comm.comm_group)
-            for iobs, obs in enumerate(data.obs):
-                log.info_rank(
-                    f"{group} : mapping observation {iobs + 1} / {len(data.obs)}.",
-                    comm=new_comm.comm_world,
-                )
-                # Data object that only covers one observation
-                obs_data = data.select(obs_uid=obs.uid)
-                # Replace comm_world with the group communicator
-                obs_data._comm = new_comm
-                job_ops.mapmaker.name = f"{orig_name}_{obs.name}"
-                job_ops.mapmaker.reset_pix_dist = True
-                job_ops.mapmaker.apply(obs_data)
+        # See if user wants separate detector maps
+        if hasattr(otherargs, "detmaps") and otherargs.detmaps:
+            my_dets = data.all_local_detectors()
```

Comment:
This might cause problems if some detectors are cut in all observations.  You could replace this line with
```
my_dets = data.all_local_detectors(flagmask=defaults.det_mask_invalid)
```
to get the global detectors which are not cut by major issues.